app/conode/stamp/stamp.go

```diff
@@ -0,0 +1,448 @@
+/*
+* Stamp - works together with a cothority-tree to sign a file. It can also verify
+* that a signature is valid.
+*
+* # Signature
+* For use in signature, run
+* ```./stamp -stamp <file>```
+* It will connect to the stampserver running on the localhost. If you want to
+* connect to another stampserver, you can give the address with the ```-server```
+* argument.
+* At the end a file signature.sig will be generated which holds all necessary
+* information necessary to check the signature.
+*
+* # Verification
+* If you want to verify whether a file is correctly signed, you can run
+* ```./stamp -verify <file.sig>```
+* which will tell whether the signature is valid. If the file referenced in the
+* file.sig is in the current directoy, it will also check it's hash.
+*/
+
+package main
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"github.com/codegangsta/cli"
+	"github.com/dedis/cothority/app/conode/defs"
+	"github.com/dedis/cothority/lib/app"
+	"github.com/dedis/cothority/lib/cliutils"
+	"github.com/dedis/cothority/lib/coconet"
+	dbg "github.com/dedis/cothority/lib/debug_lvl"
+	"github.com/dedis/cothority/lib/hashid"
+	"github.com/dedis/cothority/lib/proof"
+	"github.com/dedis/cothority/proto/sign"
+	"github.com/dedis/crypto/abstract"
+	"io"
+	"math/rand"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// Default config file
+const defaultConfigFile = "config.toml"
+
+// Default port where conodes listens
+const defaultPort = "2001"
+
+// extension given to a signature file
+const sigExtension = ".sig"
+
+// For the file-output we want a structure with base64-encoded strings, so it can be
+// easily copy/pasted
+type SignatureFile struct {
+	// name of the file
+	Name      string
+	// The time it has been timestamped
+	Timestamp int64
+	// hash of our file
+	Hash      string
+	// the root of the merkle tree
+	Root      string
+	// the inclusion-proof from root to the hash'd file
+	Proof     []string
+	// The signature challenge
+	Challenge string
+	// The signature response
+	Response  string
+	// The aggregated commitment used for signing
+	Commitment string
+}
+
```

Comment:
Right, we can use both ideas:
* Have a 
```NewSignature(filename string)```
* Which basically do 
```new(Signature).ReadFile(file string)```

Is that ok for you?